htlcswitch/link.go

```diff
@@ -130,11 +131,12 @@ type ChannelLinkConfig struct {
 	// DecodeHopIterator function is responsible for decoding HTLC Sphinx
 	// onion blob, and creating hop iterator which will give us next
 	// destination of HTLC.
-	DecodeHopIterator func(r io.Reader, rHash []byte) (HopIterator, lnwire.FailCode)
+	DecodeHopIterator func(r io.Reader, rHash []byte,
+		cltv uint32) (HopIterator, lnwire.FailCode)
 
 	// DecodeOnionObfuscator function is responsible for decoding HTLC
 	// Sphinx onion blob, and creating onion failure obfuscator.
-	DecodeOnionObfuscator func(r io.Reader) (ErrorEncrypter, lnwire.FailCode)
+	DecodeOnionObfuscator func(*sphinx.OnionPacket) (ErrorEncrypter, lnwire.FailCode)
```

Comment:
Similar comment here w.r.t violating the abstraction. This should possibly instead take a `HopIterator`, and then somehow convert that into an `ErrorEncryptor`. The exact method is up to you, it can either create a corresponding `ErrorEncryptor` object directly, or use some intermediate interface.